core/trino-main/src/main/java/io/trino/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1704,6 +1705,44 @@ private void analyzeFrameRangeOffset(Expression offsetValue, FrameBound.Type bou
             frameBoundCalculations.put(NodeRef.of(offsetValue), function);
         }
 
+        @Override
+        protected Type visitStaticMethodCall(StaticMethodCall node, Context context)
+        {
+            // PostgreSQL-style are syntactically ambiguous with static method calls. So, static method call semantics take precendence.
+            // A static method call is characterized by the target being an expression whose type is "type". This not yet supported
+            // as a first-class concept, so we fake it by analyzing the expression normally. If the analysis succeeds, we treat it as
+            // the target of a cast. If the analysis fails, check whether the target is an identifier matching a known type name.
+            try {
+                process(node.getTarget(), context);
+            }
+            catch (TrinoException e) {
+                // assume it might be a type name, so check if it's an identifier matching a known type
+                if (node.getTarget() instanceof Identifier target) {
```

Comment:
could there be separate grammar rule with higher precedence

```
identifier :: «type»
```
so that we don't need to do such instanceof checks in Java?

This could maybe allow us not to have `StaticMethodCall` "expression" which is actually only a transport for syntax, not a semantical thing.